ReliancePrinterHelper.js

```diff
@@ -0,0 +1,186 @@
+   * @returns {Array} Array of the dithered image pixels in RGBA order.
+   */
+  var dither = function(imageData, width, height) {
+    var pixelData = imageData.slice(0);
+    var index = 0;
+    for (var x = 0; x < height; x++) {
+      for (var y = 0; y < width; y++) {
+        var colored = pixelData.slice(index, index + 4);
+        var grayed = applyGrayscale(colored);
+        applySmoothing(pixelData, colored, grayed, y, x, width, height);
+        index += 4;
+      }
+    }
+    return pixelData;
+  };
+
+  /**
+   * @function oneBpp
+   * @param {Array} pixelData - Array containing the data of a dithered image in RGBA order with integers values between 0 and 255 inclusive.
+   * @param {Number} width - Width of the image, measured in pixels.
+   * @param {Number} height - Height of the image, measured in pixels.
+   * @returns {Array} Array of 8-bit integers representing the 1bpp conversion of the input image data.
+   */
+  var oneBpp = function(pixelData, width, height) {
+    var out = new Array(Math.ceil(width / 8) * height);
+    var tempByte = 0;
+    var pixIndex = 0;
+    var byteIndex = 0;
+    var bitIndex = 7;
+    for (var x = 0; x < height; x++) {
+      for (var y = 0; y < width; y++) {
+        if ((pixelData[pixIndex] * 0.299 + pixelData[pixIndex + 1] * 0.587 + pixelData[pixIndex + 2] * 0.114) < threshold) {
+          tempByte += 1 << bitIndex;
+        }
+        pixIndex += 4;
+        if (bitIndex === 0) {
+          out[byteIndex++] = tempByte;
+          tempByte = 0;
+          bitIndex = 7;
+        } else {
+          bitIndex--;
+        }
+      }
+      if (bitIndex != 7) {
+        out[byteIndex++] = tempByte;
+        tempByte = 0;
+        bitIndex = 7;
+      }
+    }
+    return out;
+  };
+
+  threshold = safeByteCast(threshold);          // Clamp the value for threshold to [0, 255].
+  divisor = Math.max(1, Math.round(divisor));   // The divisor must be an integer greater than 0.
+  var rowCount = matrixPattern.length;          // Number of rows of the matrix pattern.
+  var colCount = matrixPattern[0].length;       // Number of columns of the matrix pattern.
+  var matrixOffset;
+  for (var i = 0; i < colCount; i++) {          // Find first non-zero coefficient column in matrix. This 
+    if (matrixPattern[0][i] != 0) {             // value must always be in the first row of the matrix.
+      matrixOffset = i - 1;
+      break;
+    }
+  }
+
+  /**
+   * Public members
+   */
+
+  /**
+   * @function raster
+   * @param {Array} imageData - Array containing the data in RGBA order with integers values between 0 and 255 inclusive.
+   * @param {Number} width - Width of the image, measured in pixels.
+   * @param {Number} height - Height of the image, measured in pixels.
+   * @returns {Array} Array of 8-bit integers representing the 1bpp conversion of the input image data.
+   */
+   this.raster = function(imageData, width, height) {
+     var dithered = dither(imageData, width, height);

```

Comment:
It might be good to check here that the inputs are valid. e.g. non empty imageData, non 0/negative dimensions for width and height.